### ISSUE TITLE
Ventura: Eclipse Window becomes invisible when e.g. searching for something

### ISSUE THREAD

**Issue by c-hoeger** [XQuartz/XQuartz#306]
Since I updated to Ventura (13.1) the Eclipse window tunneled via ssh to a Debian Buster system turns invisible for some specific actions, e.g. when I do a search. I just installed XQuartz 2.8.4 and it shows the same behavior.

See attached screen recording:

https://user-images.githubusercontent.com/4416051/209631047-77dc0ecd-5274-4888-99ee-ea67c715e615.mov

**Comment by jeremyhu** [comment]
To be clear, is this behavior a regression in updating from Monterey to Ventura?  Can you confirm the window did not disappear in Monterey?

**Comment by c-hoeger** [comment]
> To be clear, is this behavior a regression in updating from Monterey to Ventura? Can you confirm the window did not disappear in Monterey?

Yes, I never had this kind of behavior before. I use this setup on a daily basis with every macOS version since many years.
Please let me know if you need any further information.

**Comment by jeremyhu** [comment]
Could you report this at http://bugreport.apple.com and provide the FB number here?  Given that it's a macOS regression, I'll try to find someone on the window server team to take a closer look.  Thanks.

**Comment by c-hoeger** [comment]
FB number is: 11934212

**Comment by svi-os** [comment]
Tk applications seem to suffer from the same issue:

https://core.tcl-lang.org/tk/tktview/3fa79af37156bdcb00e2ef2a1cce93a2ef8a738c

**Comment by dpeterc** [comment]
I think this bug is a variation of 
https://github.com/XQuartz/XQuartz/issues/293
since both are a regression of Ventura, regardless of XQuartz version. I am seeing this bug very often in my applications, since I use many windows/dialogs, which can be open in parallel.
In my opinion there is some problem in handing window focus. When child window (dialog on top of main window) is closed, the focus should pass to the parent widget. Somehow this child to parent focus passage goes wrong, and instead of raising the shell of the parent and passing the focus to last selected widget, the whole window is iconized / disappears. It can be retrieved by calling the function  "Window > Bring All to Front", from XQaurtz main menu, but regular user is totally confused, and thinks the application has crashed.

**Comment by jeremyhu** [comment]
It is probably related to the same architectural changed in Ventura, but it is unfortunatley not the same as #293

**Comment by c-hoeger** [comment]
JFYI: I received a response from Apple:

```
There are changes in the latest update, build 22E5219e (Ventura 13.3 Beta 1), that may have resolved this issue.
You can see the software build your device is running and check for the latest update by clicking on the Apple logo in the upper left hand corner > About This Mac. If the build is not visible, click on the macOS version, e.g. 10.15.x, to reveal it.

Has this issue been resolved after installing the latest update?
```

Unfortunately I don't have a second machine were I could test it.
So if anybody else wants to test.

**Comment by dpeterc** [comment]
I have installed the latest Ventura 13.3 Beta (22E5219e) on iMac 27" Intel i9, with latest XQuartz 2.8.5, and I am sorry to report that the problem has not been solved.
At least on my Motif based application, when closing a dialog window, the focus it not correctly transferred to the parent window, and the whole application disappears. It can be restored with"Window > Bring All to Front", but to the average user it appears crashed.

**Comment by c-hoeger** [comment]
Meanwhile I installed the Beta 13.3 on a different computer (also an iMac 27) and can confirm, that the problem isn't solved. Will send that feeback to Apple.

**Comment by jeremyhu** [comment]
@dpeterc That is not what this bug is about.  Also, I don't think anyone has filed a bug report about that issue.

**Comment by jeremyhu** [comment]
@c-hoeger Correct.  This issue is not solved in the latest betas.  It was not expected to be.

**Comment by dpeterc** [comment]
I have posted a new ticket with videos with simple open source software, so it is easier to track the problem
https://github.com/XQuartz/XQuartz/issues/329

**Comment by jeremyhu** [comment]
This issue should now be addressed in macOS 13.3 Beta 2.  I have not verified that though.

**Comment by dpeterc** [comment]
I got the notification with release notes about macOS 13.3 Beta 2, but can not download it. System setting say that this Mac is enrolled in Beta program, but claims that I am up to date (with previous Beta).
Has it not been released on Intel Mac's yet? Or is there a gradual rollout by region. I am in Europe.

**Comment by c-hoeger** [comment]
same here...

**Comment by dpeterc** [comment]
I have now managed to upgrade to newer version of macOS 13.3 beta (22E5230e) - it does not say Beta 2.
The application no longer disappears on closing the child window.

**Comment by c-hoeger** [comment]
Ack, was now able to verify as well. The reported issue is fixed.
